File: python/leetcode/442. Find All Duplicates in an Array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        ans = []
        for i in range(len(nums)):
            val = nums[i]
            if val < 0:
                val *= -1

            value = nums[val - 1]

            if value < 0:
                ans.append(val)
            else:
                nums[val - 1] = -value
        return ans

File: c++/leetcode/442. Find All Duplicates in an Array.cpp
class Solution
{
public:
  vector<int> findDuplicates(vector<int> &nums)
  {
    vector<int> result;

    for (int i = 0; i < nums.size(); i++)
    {
      int val = nums[i];
      val = val < 0 ? -val : val;

      int value = nums[val - 1];

      if (value < 0)
      {
        result.push_back(val);
      }
      else
        nums[val - 1] = -value;
    }

    return result;
  }
};
